model/map/src/main/java/org/keycloak/models/map/userSession/MapAuthenticatedClientSessionEntity.java

```diff
@@ -16,184 +16,80 @@
  */
 package org.keycloak.models.map.userSession;
 
-import org.keycloak.common.util.Time;
+import org.keycloak.models.map.annotations.GenerateEntityImplementations;
 import org.keycloak.models.map.common.AbstractEntity;
 
+import org.keycloak.models.map.common.DeepCloner;
 import org.keycloak.models.map.common.UpdatableEntity;
 import java.util.Map;
-import java.util.Objects;
-import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * @author <a href="mailto:mkanis@redhat.com">Martin Kanis</a>
  */
-public class MapAuthenticatedClientSessionEntity extends UpdatableEntity.Impl implements AbstractEntity {
+@GenerateEntityImplementations(
+        inherits = "org.keycloak.models.map.userSession.MapAuthenticatedClientSessionEntity.AbstractAuthenticatedClientSessionEntity"
+)
+@DeepCloner.Root
+public interface MapAuthenticatedClientSessionEntity extends AbstractEntity, UpdatableEntity {
 
-    private String id;
-    private String userSessionId;
-    private String realmId;
-    private String clientId;
+    public abstract class AbstractAuthenticatedClientSessionEntity extends UpdatableEntity.Impl implements MapAuthenticatedClientSessionEntity {
 
-    /**
-     * Flag signalizing that any of the setters has been meaningfully used.
-     */
+        private String id;
 
-    private String authMethod;
-    private String redirectUri;
-    private volatile int timestamp;
-    private long expiration;
-    private String action;
+        @Override
+        public String getId() {
+            return this.id;
+        }
 
-    private Map<String, String> notes = new ConcurrentHashMap<>();
+        @Override
+        public void setId(String id) {
+            if (this.id != null) throw new IllegalStateException("Id cannot be changed");
+            this.id = id;
+            this.updated |= id != null;
+        }
 
-    private String currentRefreshToken;
-    private int currentRefreshTokenUseCount;
-
-    private boolean offline;
-
-    public MapAuthenticatedClientSessionEntity() {}
-
-    public MapAuthenticatedClientSessionEntity(String id, String userSessionId, String realmId, String clientId, boolean offline) {
-        this.id = id;
-        this.userSessionId = userSessionId;
-        this.realmId = realmId;
-        this.clientId = clientId;
-        this.offline = offline;
-        this.timestamp = Time.currentTime();
+        @Override
+        public String toString() {
```

Comment:
this should not be here, we want to generate this so it is consistent across all entities.

Reply:
If I recall it correctly this format was useful when debugging session/login failure entities (already merged in login failures). Should we use default format `getClass().getName() + "@" + Integer.toHexString(hashCode())` anyway?

Reply:
`toString` method is generated by our generator to something similar you use here. The advantage is, that with the generator we can change it in one place and it will be changed in each entity.

See: https://github.com/keycloak/keycloak/blob/main/model/build-processor/src/main/java/org/keycloak/models/map/processor/GenerateEntityImplementationsProcessor.java#L336-L338

And: https://github.com/keycloak/keycloak/blob/main/model/build-processor/src/main/java/org/keycloak/models/map/processor/GenerateHotRodEntityImplementationsProcessor.java#L212-L214

Reply:
Thank you for pointing me to this. I removed it also from `userLoginFailures`.